From 2d69c42106c1e6791c174b289ad94bcc667bde33 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <matthias.schiffer@tq-group.com>
Date: Fri, 28 Apr 2023 10:39:37 +0200
Subject: [PATCH] iMX8M: add mkimage_imx8 wrapper to redirect output

Add a wrapper that tees the stderr output to a log file, allowing for
more robust parsing for CSF / HAB block offsets.

Signed-off-by: Matthias Schiffer <matthias.schiffer@tq-group.com>
---
 iMX8M/mkimage_imx8 | 5 +++++
 iMX8M/soc.mak      | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)
 create mode 100755 iMX8M/mkimage_imx8

diff --git a/iMX8M/mkimage_imx8 b/iMX8M/mkimage_imx8
new file mode 100755
index 0000000..4810d13
--- /dev/null
+++ b/iMX8M/mkimage_imx8
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+OUTLOG=mkimage.log
+
+exec ./mkimage_imx8.real "$@" 2>&1 | tee "${OUTLOG}" >&2
diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index a98195e..b43061c 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -1,4 +1,4 @@
-MKIMG = mkimage_imx8
+MKIMG = mkimage_imx8.real
 OUTIMG = flash.bin
 
 CC ?= gcc
